### PR TITLE
Move ToolManager warnings to rcParam validator

### PR DESCRIPTION
**Pull request by EBenkler**
## PR Summary

Applies patch from [#15284](https://github.com/matplotlib/matplotlib/issues/15284#issuecomment-687015853)
See also [#19458](https://github.com/matplotlib/matplotlib/pull/19458#issue-568076935)

## PR Checklist

- [x] Has pytest style unit tests (and `pytest` passes).
- [x] Is [Flake 8](https://flake8.pycqa.org/en/latest/) compliant (run `flake8` on changed files to check).
- [ ] New features are documented, with examples if plot related.
- [ ] Documentation is sphinx and numpydoc compliant (the docs should [build](https://matplotlib.org/devel/documenting_mpl.html#building-the-docs) without error).
- [ ] Conforms to Matplotlib style conventions (install `flake8-docstrings` and run `flake8 --docstring-convention=all`).
- [ ] New features have an entry in `doc/users/next_whats_new/` (follow instructions in README.rst there).
- [ ] API changes documented in `doc/api/next_api_changes/` (follow instructions in README.rst there).

### CODE REVIEW

File: lib/matplotlib/rcsetup.py [hunk 1]
```diff
@@ -309,6 +309,15 @@ def validate_backend(s):
...
     _deprecated_since="3.3")
 
+def _validate_toolbar(s):
+    s = ValidateInStrings(
+        'toolbar', ['None', 'toolbar2', 'toolmanager'], ignorecase=True)(s)
+    if s == 'toolmanager':
+        cbook._warn_external(
+            "Treat the new Tool classes introduced in v1.5 as experimental "
+            "for now, the API will likely change in version 2.1 and perhaps "
+            "the rcParam as well.")
```
Comment (QuLogic) [inline lib/matplotlib/rcsetup.py:319]:
```suggestion
            "for now, the API and rcParam may change in future versions.")
```

**Review by QuLogic — COMMENTED** [review]
You need to fix the flake8 errors as well.

File: lib/matplotlib/rcsetup.py [hunk 2]
```diff
@@ -310,6 +310,16 @@ def validate_backend(s):
     _deprecated_since="3.3")
 
 
+def _validate_toolbar(s):
+    s = ValidateInStrings(
+        'toolbar', ['None', 'toolbar2', 'toolmanager'], ignorecase=True)(s)
+    if s == 'toolmanager':
+        cbook._warn_external(
```
Comment (QuLogic) [inline lib/matplotlib/rcsetup.py:317]: Should be 
```suggestion
        _api.warn_external(
```

Comment (EBenkler) [reply lib/matplotlib/rcsetup.py]: Yes, I already had realized the cbook / _api issue, but I also had to adjust the tests
(test_backend_bases.py::test_toolbar_zoompan, please revise if the changes are ok)

Now, pytest passes, but an Azure Pipeline fails. I don't know what I can do about that and if it is a serious issue. Any chance you have a look at it?

**Review by anntzer — APPROVED** [review]
the azure failure is irrelevant.

**Review by QuLogic — APPROVED** [review]

**Comment by QuLogic** [comment]
Thanks @EBenkler! Congratulations on your first PR to Matplotlib :tada: We hope to hear from you again.

**Pull request merged**